server/lib/renderers/xlsx.js

```diff
@@ -3,34 +3,103 @@
  *
  * @description
  * This library is just a shim to ensure API uniformity with other renderers.
- * The core functionality is identical to the JSON renderer.
- *
+ * it renders an Excel report(from data.rows that we pass as render function param).
+ * Having the same data structure will help to have less xlsx rendereres
  * @module lib/renderers/xlsx
  *
  * @requires q
  */
-
-const q = require('q');
+const xl = require('excel4node');
+const _ = require('lodash');
 
 const headers = {
-  'Content-Type' : 'application/xlsx',
-  type : 'xlsx',
+  'Content-Type' : 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
 };
 
-exports.render = renderXLSX;
+exports.render = render;
 exports.extension = '.xlsx';
 exports.headers = headers;
 
 /**
- * JSON Render Method
- * @param {Object} data      Contains data in any format that will be used to
- *                            drive the final report - this will be exposed to
- *                            the view by the renderer.
- * @param {Object} options   (Optional) parameters that can be passed to switch
- *                            render features on/off.
- * @returns {Object}          JSON Object representing a report that can be sent
- *                            to the client.
+ * XLSX Render Method
+ * @param {Object} data   { rows : []}
  */
-function renderXLSX(data) {
-  return q.resolve(data);
+
+function render(data) {
+
+
+  // Create a new instance of a Workbook class
+  const wb = new xl.Workbook();
+
+  // style applied to cells
+  const headerStryle = wb.createStyle({
+    font : {
+      color : '#ffffff',
+      size : 12,
+      bold : true,
+    },
+    fill : {
+      type : 'pattern',
+      patternType : 'solid',
+      fgColor : '2384F5',
+    },
+  });
+
+  const normaFontSize = {
+    font : {
+      size : 12,
+    },
+  };
+  const borderInfo = {
+    style : 'thin',
+    color : '#000000',
+  };
+
+  const styleAllBorders = {
+    border : {
+      bottom : borderInfo,
+      left : borderInfo,
+      top : borderInfo,
+      right : borderInfo,
+    },
+  };
+  // Add Worksheets to the workbook
+  const ws = wb.addWorksheet('Sheet 1');
+  const { rows } = data;
+  const firstObject = rows[0] || {};
+
+  // writing columns
+  const firstLineCols = Object.keys(firstObject);
+  let line = 1;
+  firstLineCols.forEach((key, index) => {
+    ws.cell(line, index + 1).string(key)
```

Comment:
For consistency, could you put the `.string(key)` on the next line?  It should look like this:
```js
ws.cell(line,index + 1)
  .string(key)
  .style(headerStyle)
  .style(styleAllBorders);
```